lib/Dialect/AIEVec/IR/AIEVecOps.cpp

```diff
@@ -389,30 +394,40 @@ void aievec::FMAOp::print(OpAsmPrinter &p) {
 // Verify Mul and FMA op.
 template <typename T> LogicalResult verifyMulFMAOp(T op) {
   // Verify the types
-  aievec::AccType resultType =
-      op.result().getType().template dyn_cast<aievec::AccType>();
   VectorType lhsType = op.lhs().getType().template dyn_cast<VectorType>();
   VectorType rhsType = op.rhs().getType().template dyn_cast<VectorType>();
 
   if (!lhsType || !rhsType)
     return op.emitError("requires vector type");
-  if (!resultType)
+
+  // Determine if it is an integer or float operation
+  bool isInt = lhsType.getElementType().isa<IntegerType>();
+
+  VectorType resultVecType =
+      op.result().getType().template dyn_cast<VectorType>();
+  aievec::AccType resultAccType =
+      op.result().getType().template dyn_cast<aievec::AccType>();
+  if (isInt && !resultAccType)
     return op.emitError("requires accumulator type");
+  else if (!isInt && !resultVecType)
+    return op.emitError("requires vector type");
```

Comment:
It seems like these checks should just be type constraints on the op?

Reply:
Huh, yeah it would seem that these sorts of checks should already be handled by the tablegen constraints: I'm seeing them all over the file. I'll take a look to see if these sorts of checks actually catch anything the tablegen constraints don't. Would clearing out these dead type checks constitute another issue/PR?

Reply:
Yes, handling it separately is fine.  Having logically atomic and separately verifiable changes is good practice.